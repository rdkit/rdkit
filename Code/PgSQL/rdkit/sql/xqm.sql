select 'Cc1[nH]nc(F)c1'::mol @> mol_to_tautomerquery('Cc1n[nH]c(F)c1'::mol);
select substruct('Cc1[nH]nc(F)c1'::mol,mol_to_tautomerquery('Cc1n[nH]c(F)c1'::mol));
select substruct('Cc1[nH]nc(F)c1'::mol,mol_to_tautomerquery('Cc1n[nH]c(F)n1'::mol));

select substruct('COC'::mol,'COC |LN:1:1.3|'::mol);
select substruct('COC'::mol,mol_enumeratequery('COC |LN:1:1.3|'::mol));
select substruct('COOC'::mol,'COC |LN:1:1.3|'::mol);
select substruct('COOC'::mol,mol_enumeratequery('COC |LN:1:1.3|'::mol));

-- error handling
select mol_to_tautomerquery('C1'::mol);
select mol_to_tautomerquery('c1cc1'::mol);
