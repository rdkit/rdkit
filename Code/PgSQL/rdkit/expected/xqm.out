select 'Cc1[nH]nc(F)c1'::mol @> mol_to_tautomerquery('Cc1n[nH]c(F)c1'::mol);
 ?column? 
----------
 t
(1 row)

select substruct('Cc1[nH]nc(F)c1'::mol,mol_to_tautomerquery('Cc1n[nH]c(F)c1'::mol));
 substruct 
-----------
 t
(1 row)

select substruct('Cc1[nH]nc(F)c1'::mol,mol_to_tautomerquery('Cc1n[nH]c(F)n1'::mol));
 substruct 
-----------
 f
(1 row)

select substruct('COC'::mol,'COC |LN:1:1.3|'::mol);
 substruct 
-----------
 t
(1 row)

select substruct('COC'::mol,mol_enumeratequery('COC |LN:1:1.3|'::mol));
WARNING:  create bndl 3
WARNING:  depickle bndl 3
WARNING:  bndl 3
 substruct 
-----------
 t
(1 row)

select substruct('COOC'::mol,'COC |LN:1:1.3|'::mol);
 substruct 
-----------
 f
(1 row)

select substruct('COOC'::mol,mol_enumeratequery('COC |LN:1:1.3|'::mol));
WARNING:  create bndl 3
WARNING:  depickle bndl 3
WARNING:  bndl 3
 substruct 
-----------
 t
(1 row)

-- error handling
select mol_to_tautomerquery('C1'::mol);
ERROR:  could not create molecule from SMILES 'C1'
LINE 1: select mol_to_tautomerquery('C1'::mol);
                                    ^
select mol_to_tautomerquery('c1cc1'::mol);
ERROR:  MolToTautomerQuery: Can't kekulize mol.  Unkekulized atoms: 0 1 2
